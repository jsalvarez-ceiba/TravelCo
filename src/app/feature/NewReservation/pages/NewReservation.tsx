import React from 'react';
import FormReservation from '../components/FormReservation/FormReservation';
import './NewReservation.style.scss';
import Footer from '../../Footer/components/Footer';



const NewReservation = () => {
    return (
        
        <div className="background">
            <div className="d-flex justify-content-center" style={{padding: 20}}>
                <i className="fas fa-plane-departure fa-2x text-white"></i>
            </div>
            <FormReservation places={[]}  />
            <Footer />
        </div>
        
    );
};
export default NewReservation;