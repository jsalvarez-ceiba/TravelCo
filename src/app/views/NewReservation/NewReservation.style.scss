.backgroundFly {
    background-image: url('../../../assets/img/travel.jpg');
    height: auto;
    
}