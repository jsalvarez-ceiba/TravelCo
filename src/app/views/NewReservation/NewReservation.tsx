import React from 'react';
import './NewReservation.style.scss';
import FormReservation from '../../components/FormReservation/FormReservation';

const NewReservation = () => {
    return (
        <div className="background">
            
            <FormReservation />
        </div>
    );
};

export default NewReservation;
