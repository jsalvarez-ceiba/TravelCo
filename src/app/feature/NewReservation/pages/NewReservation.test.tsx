import React from 'react';
import { mount } from 'enzyme';
import { render } from '@testing-library/react';
import {prettyDOM} from '@testing-library/dom';
import NewReservation from './NewReservation';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';



const middlewares = [thunk];
const mockStore = configureMockStore(middlewares)




describe('Render Home View', () => {

  /* const initialState = {}; */
  /* const mockStore = configureMockStore(); */
  let store, wrapper;

  test('render component', () => {
    store = mockStore({});
    const component = mount(
      <Provider store={store}>
        <NewReservation />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('backgroundFlyCell', () => {
    store = mockStore({});
    const component = render(
      <Provider store={store}>
        <NewReservation />
      </Provider>
    );

    const div = component.container.querySelector('.backgroundFlyCell');

    /* console.log(prettyDOM(div)); */

     expect(div).not.toBeNull(); 


  })

});
