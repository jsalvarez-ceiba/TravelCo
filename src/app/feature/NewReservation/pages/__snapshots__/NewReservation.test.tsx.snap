// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home View render component 1`] = `
<div
  className="background"
>
  <div
    className="d-flex justify-content-center"
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <i
      className="fas fa-plane-departure fa-2x text-white"
    />
  </div>
  Array [
    <div
      className="card mx-auto opacityCell"
      data-aos="zoom-in"
    >
      <div
        className="card-header"
      >
        <div
          className=""
        >
          <strong>
            Nueva Reservación
          </strong>
        </div>
      </div>
      <div
        className="card-body"
      >
        <h6
          className="card-title"
        >
          Información del vuelo
        </h6>
        <hr />
        <div
          className="form-group"
        >
          <strong>
            Origen
          </strong>
          <select
            className="form-control"
            name="origin"
            onChange={[Function]}
          >
            <option
              value=""
            >
              Seleccionar 
              Origen
               
            </option>
          </select>
        </div>
        <br />
        <div
          className="form-group"
        >
          <strong>
            Destino
          </strong>
          <select
            className="form-control"
            name="destination"
            onChange={[Function]}
          >
            <option
              value=""
            >
              Seleccionar 
              Destino
               
            </option>
          </select>
        </div>
        <br />
        <div
          className="form-group"
        >
          <strong>
            Fecha y Hora
          </strong>
          <input
            className="form-control"
            max="2022-08-25"
            name="date"
            onChange={[Function]}
            type="date"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="fltime"
            onChange={[Function]}
            type="time"
            value=""
          />
        </div>
        <br />
        <div
          className="form-group"
        >
          <strong>
            Precio viaje : 
            0
             COP 
          </strong>
        </div>
        <div
          className="form-group"
        />
        <div
          className="form-group"
        >
          <strong>
            Total: 
            0
             
          </strong>
        </div>
      </div>
    </div>,
    <div
      className="card mx-auto opacityCell"
      data-aos="zoom-in"
    >
      <div
        className="card-body"
      >
        <h6
          className="card-title"
        >
          Datos personales 
        </h6>
        <hr />
        <div
          className="form-group"
        >
          <strong>
            Nombres
          </strong>
          <input
            className="form-control"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <br />
        <div
          className="form-group"
        >
          <strong>
            Apellidos
          </strong>
          <input
            className="form-control"
            name="lastname"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <br />
        <div
          className="form-group"
        >
          <strong>
             Fecha de Nacimiento 
          </strong>
          <input
            className="form-control"
            name="birthdate"
            onChange={[Function]}
            type="date"
            value=""
          />
        </div>
        <br />
        <strong
          className="text-danger"
        >
          Debes ser mayor de edad para generar una reserva!
        </strong>
      </div>
    </div>,
    "",
  ]
  <div
    className="flexCenter bg-dark"
  >
    <small
      className="text-white"
    >
      TravelCo | Todos los derechos reservados 2021
    </small>
  </div>
</div>
`;
